.github/workflows/code_checks.yml

```diff
@@ -68,10 +68,72 @@ jobs:
           number: ${{ github.event.pull_request.number }}
           labels: lint-failure
 
+  build:
+    name: Build and Cache Docker Image
+    runs-on: ubuntu-16-cores-latest
+    env:
+      BUNDLE_ENTERPRISE__CONTRIBSYS__COM: ${{ secrets.BUNDLE_ENTERPRISE__CONTRIBSYS__COM }}
+      DOCKER_BUILDKIT: 1
+    steps:
+      - name: Checkout PR
+        if: ${{ github.event_name != 'pull_request' }}
+        uses: actions/checkout@v5
+
+      - name: Checkout PR HEAD ref
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: actions/checkout@v5
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v5
+        with:
+          role-to-assume: ${{ vars.AWS_ASSUME_ROLE }}
+          aws-region: us-gov-west-1
+
+      - name: Login to Amazon ECR
+        id: ecr-login
+        uses: aws-actions/amazon-ecr-login@v2.0.1
+        with:
+          mask-password: true
+
+      - name: Set Environment Variable
+        run: echo "VETS_API_USER_ID=$(id -u)" >> $GITHUB_ENV
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and Cache Docker Image
+        uses: docker/build-push-action@v6
+        env:
+          DOCKER_BUILD_SUMMARY: false
+        with:
+          build-args: |
+            BUNDLE_ENTERPRISE__CONTRIBSYS__COM=${{ env.BUNDLE_ENTERPRISE__CONTRIBSYS__COM }}
+            USER_ID=${{ env.VETS_API_USER_ID }}
+          context: .
+          push: false
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
   tests:
-    name: Test
-    runs-on: ubuntu-32-cores-latest
+    name: Test (Group ${{ matrix.test_group }})
+    runs-on: ubuntu-16-cores-latest
```

Comment:
Drop to 16 cores since only 1 process will be running instead of 24 (could possible drop to 8, would need to benchmark).